server/rpc/filters/filters.go

```diff
@@ -209,13 +210,15 @@ func GetStreamInterceptor(env environment.Env) grpc.ServerOption {
 
 func GetUnaryClientInterceptor() grpc.DialOption {
 	return grpc.WithChainUnaryInterceptor(
+		otelgrpc.UnaryClientInterceptor(),
 		grpc_prometheus.UnaryClientInterceptor,
 		setHeadersUnaryClientInterceptor(),
 	)
 }
 
 func GetStreamClientInterceptor() grpc.DialOption {
 	return grpc.WithChainStreamInterceptor(
+		otelgrpc.StreamClientInterceptor(),
```

Comment:
do these have any effect when tracing is disabled?

Reply:
Yes, there is a small amount of overhead in creating the trace info even when tracing is disabled or if a trace is not sampled. 
I don't think it will have a noticeable impact on performance. We can do some perf testing next week to make sure.